.github/workflows/cryostat-IT.yml

```diff
@@ -0,0 +1,121 @@
+on:
+  workflow_call:
+    inputs:
+      build-arch:
+        required: true
+        type: string
+      skip-itests:
+        required: false
+        type: boolean
+        default: false
+    outputs:
+      image-version:
+        description: the Cryostat application version that will be built
+        value: ${{ jobs.get-pom-properties.outputs.image-version }}
+
+jobs:
+  get-pom-properties:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        repository: ${{ github.repository_owner }}/cryostat
+        ref: main
+    - uses: Mudlet/xmlstarlet-action@master
+      with:
+        args: ed -P -L -u '//_:properties[_:io.cryostat.core.version="2.23.0"]/_:io.cryostat.core.version' -v "2.24.0-SNAPSHOT" pom.xml 
+    - id: query-pom
+      name: Get properties from POM
+      # Query POM for image version and save as output parameter
+      run: |
+        IMAGE_VERSION="$(mvn validate help:evaluate -Dexpression=cryostat.imageVersionLower -q -DforceStdout)"
+        echo "image-version=$IMAGE_VERSION" >> "$GITHUB_OUTPUT"
+    outputs:
+      image-version: ${{ steps.query-pom.outputs.image-version }}
+
+  build-image:
+    runs-on: ubuntu-latest
+    needs: [get-pom-properties]
+    steps:
+    - name: Install qemu
+      if: ${{ inputs.build-arch != 'amd64' }}
+      continue-on-error: true
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y qemu-user-static
+    - uses: actions/checkout@v4
+      if: always()
+      with:
+        repository: ${{ github.repository_owner }}/cryostat
+        ref: main
+        submodules: true
+        fetch-depth: 0
+    - uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+    - name: maven-settings
+      uses: s4u/maven-settings-action@v2
+      with:
+        servers: '[{"id": "github", "username": "dummy", "password": "${env.GITHUB_TOKEN_REF}"}]'
+        githubServer: false
+    - uses: skjolber/maven-cache-github-action@v1
+      with:
+        step: restore
+    - run: git submodule init
+    - run: git submodule update
+    - uses: actions/download-artifact@v3
+      with:
+        name: cryostat-core
```

Comment:
I am not sure if this worked. This will download the `-core` artifact in local repository but not in local maven repo. Even so, not sure if maven knows about it when manually unpacking this artifact into local maven repo.

Reply:
We will need to cache the entire local maven repo when building and saving `-core` artifact. Can we use https://github.com/actions/setup-java#caching-packages-dependencies?

Reply:
isn't  `skjolber/maven-cache-github-action@v1` caching the maven dependencies?

Reply:
Hmm, actually from a closer look: 

https://github.com/skjolber/maven-cache-github-action#cache-scopes

Reply:
Looks like the cache is scoped to a specific branch. So PR's workflow runs won't conflict.

Reply:
Perhaps, its the same pattern for caching. So, `setup-java` should have cache scoped to branches too.

Reply:
![image](https://github.com/cryostatio/cryostat-core/assets/68053619/3b1582d1-5742-46e4-bd86-44cd0701e427)